spec/requests/posts_spec.rb

```diff
@@ -1,6 +1,32 @@
 require 'rails_helper'
 
 RSpec.describe 'Posts', type: :request do
+  let(:user) do
+    User.create(
+      id: 1,
+      name: 'Mike Patton',
+      photo: 'https://images.unsplash.com/photo-1525134479668-1bee5c7c6845?ixlib=rb-4.0.3&amp;ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&amp;auto=format&amp;fit=crop&amp;w=687&amp;q=80',
+      bio: 'I am a musician',
+      posts_counter: 3
+    )
+  end
+
+  let(:post) do
+    Post.create(
+      id: 1,
+      title: 'My First Post',
+      text: 'This is my first post',
+      author: user,
+      comments_counter: 0,
+      likes_counter: 0
+    )
+  end
+
+  before do
+    user
+    post
+  end
+
```

Comment:
- [ ] You did a good job working on the previous reviewer's suggestions! 💯 However, I'm getting a new error:

![image](https://github.com/smunoz1988/blog-on-rails/assets/77069011/adbb6c6b-4e56-49e3-a4d3-b6992bb8f818)

This happens because you're explicitly using the id to create a User, and somewhere in your test suite, there's already one with this id. Instead, I would suggest using a more dynamic approach, something like this:

Define variables `user` and `post`:
```rb
  let(:user) do
    User.create(

      name: 'Mike Patton',
      photo: 'https://images.unsplash.com/photo-1525134479668-1bee5c7c6845?ixlib=rb-4.0.3&amp;ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&amp;auto=format&amp;fit=crop&amp;w=687&amp;q=80',
      bio: 'I am a musician',
      posts_counter: 3
    )
  end

  let(:post) do
    Post.create(

      title: 'My First Post',
      text: 'This is my first post',
      author: user,
      comments_counter: 0,
      likes_counter: 0
    )
  end
```

and then, in your routes, use the ids created automatically by the database instead of using fixed numbers like `1` or `514`.

```rb
get "/users/#{user.id}/posts"

get "/users/#{user.id}/posts/#{post.id}"
```

Please consider applying this change! 🤓

Reply:
@anagudelogu thanks for the recommendations! 

fixed!